astropy/io/fits/column.py

```diff
@@ -1389,16 +1387,13 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
-            attr = []
-            for col in self:
-                val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+            attr = [getattr(col, cname) for col in self.columns]
+            if None in attr:
```

Comment:
From what I can see a list comprehension is faster than a loop with append (~2x). And yes, I guess the benefit comes from the fact that most of the time `attr` is not `None`, at least in my case, but it is difficult to say exactly as it will depend on the tables. 
If you look at the timings I gave in the first post, this is the change that brings the better speedup, so there is clearly something to do for this function, but I don't know if there is a better solution.